aiorospy/src/aiorospy/topic.py

```diff
@@ -1,18 +1,42 @@
+import asyncio
 import janus
 import rospy
 
+from functools import partial
 
-class AsyncSubscriber:
 
-    def __init__(self, name, data_class):
-        self._queue = janus.Queue()
-        self._subscriber = rospy.Subscriber(name, data_class, lambda msg: self._queue.sync_q.put(msg))
+class AsyncSubscriber:
+    def __init__(self, name, data_class, queue_size=None, loop=None):
+        self.name = name
+        self._data_class = data_class
+        self._queue_size = queue_size
+        self._loop = loop if loop is not None else asyncio.get_running_loop()
 
-        self.unregister = self._subscriber.unregister
+    # TODO(pbovbel) should we check rospy.is_shutdown() instead of while True?
```

Comment:
It doesn't matter, I don't think.  Because when we use asyncio + rospy we have to (in my experience) set `disable_signals=True`. Which means that this loop will be interrupted before the shutdown signal is set.  I haven't tested it, but my guess is that doing so would leave us worse off.

Reply:
I agree that it probably doesn't matter, but I'll dig a little deeper. It feels wrong to keep waiting for a message that never comes.